### PR TITLE
Course model, email confirmation and course administration

### PR DESCRIPTION
**Pull request by jnsdrtlf**
This branch contains many changes to the authentication system behind GewässerCampus. A new model called `Course` has been introduced that contains course data previously hold by a `CourseToken`. The models `CourseToken` and `AccessKey` now inherit from the same `BaseToken` and have permissions.

Other new features:

- [x] Access the current token user with `request.token`
- [x] No more hidden token fields in the measurement forms. Permissions are used instead.
- [x] Stateless (almost) email confirmation
- [x] Emails now no longer require a plain text template. The plain text is extracted using pandoc (`pypandoc` installed with `conda`).
- [x] Updated course administration

Todo:

- [x] Update (nearly all) tests. The mechanics changed fundamentally and many tests are no longer needed or need updates.
- [x] Add button to deactivate and copy access keys in the course administration

### CODE REVIEW

**Comment by jnsdrtlf** [comment]
Will fix #57

**Review by ax-sc — COMMENTED** [review]
Looks good to me overall, had just some minor comments. Should we merge `develop` into `main` first and release version 0.3.4 before merging this PR?

File: gcampus/auth/recievers.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        logger.debug("Access key has been created, skip updating 'CourseOverviewPDF'.")
+        return
+    course: Course = instance.course
+    render_cached_document_view.apply_async(
+        args=(
+            "gcampus.documents.views.CourseOverviewPDF",
+            course.pk,
+            get_language(),
+        ),
+    )
```
[the remark this answers is not in the document]
Comment (jnsdrtlf) [reply gcampus/auth/recievers.py]: Fixed with 4764df4

File: environment.yml [hunk 2]
```diff
@@ -8,5 +8,6 @@ dependencies:
   - libgdal=3.4
   - glib=2.70
   - pango=1.50
+  - pandoc=2.18
```
[the remark this answers is not in the document]
Comment (jnsdrtlf) [reply environment.yml:11]: Added instructions to the documentation in 4206ff4

**Review by ax-sc — APPROVED** [review]

**Pull request merged**